app/tournament/tournament.js

```diff
@@ -0,0 +1,93 @@
+import {Player, validatePlayer} from "./player";
+import {Match, ChildMatch} from "./match";
+import * as utils from "../common/utils";
+
+export class Tournament {
+    constructor() {
+        this.players = [];
+        this.matches = [];
+    }
+
+    addPlayer(player) {
+        this.players.push(new Player(player.name, player.rank));
```

Comment:
Just in case the parameter was an object containing name and rank instead of an instance of Player.